### ISSUE TITLE
Bubbles in water trend to right.

### ISSUE THREAD

**Issue by vi** [MaxBittker/sandspiel#62]
Make a bowl using Wall element, fill it with Water, then make a small hole at the bottom with Empty.

I expect bubbles to float up symmetrically, just like water is falling down. Instead they strongly tend to right.

If it is due to processing scanline direction when calculating, then this direction may be alternated bettween even and odd frames to negate the effect.

**Comment by MaxBittker** [comment]
you're correct that this is caused by the scanline - I think that alternating gave me a performance regression when I tried it, but there may be a good solution (someone had suggested that the vertical iteration order is not efficient in the past so there may be other gains which could offset this)
